### PR TITLE
Silicon Flavortexts. (SCRUTINISE HEAVILY)

### PR DESCRIPTION
**Pull request by Vizoplez**
## About The Pull Request
Does as the title says, gives silicons the ability to have a seperate flavortext, but keep the same OOC shit.

## Why It's Good For The Game
This shouldve been a feature from the start, silicons being unable to show their ooc desires (e.g not wanting to erp) shouldve been a high priority feature from the start.

## Changelog
:cl:
add: The ability for silicons to have flavortexts (borgs,drones,etc)
server: CHANGES PREF SAVING (HEAVILY REVIEW THIS)
/:cl:

### CODE REVIEW

**Comment by Yawet330** [comment]
The only issues currently are as follows:
-The pref not saving reliably

**Comment by Vizoplez** [comment]
Prefs now working, still failing to apply roundstart ingame, but can be changed and work once ingame.
image

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -367,6 +368,15 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 					dat += "[features["flavor_text"]]"
 			else
 				dat += "[TextPreview(features["flavor_text"])]...<BR>"
+			dat += "<BR>"
```
Comment (silicons) [inline code/modules/client/preferences.dm]: use the same format as above please, h2 instead of br, it'll look weird otherwise

File: code/modules/client/preferences_savefile.dm [hunk 2]
```diff
@@ -553,6 +553,15 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 	else //We have no old flavortext, default to new
 		S["feature_flavor_text"]		>> features["flavor_text"]
 
+	if((S["silicon_flavor_text"] != "") && (S["silicon_flavor_text"] != null) && S["silicon_flavor_text"]) // dont change what isnt broken
```
Comment (silicons) [inline code/modules/client/preferences_savefile.dm]: just do S["silicon_flavor_text"] >> features["silicon_flavor_text"]

Comment (silicons) [reply code/modules/client/preferences_savefile.dm]: the whole conversion thing isn't needed since this is a new thing not an old thing that was converted

File: code/modules/mob/living/silicon/silicon.dm [hunk 3]
```diff
@@ -56,6 +56,14 @@
 	diag_hud_set_status()
 	diag_hud_set_health()
 
+
```
Comment (silicons) [inline code/modules/mob/living/silicon/silicon.dm]: nitpick but no extra line here

File: code/modules/mob/living/silicon/silicon.dm [hunk 4]
```diff
@@ -56,6 +56,14 @@
...
+/mob/living/silicon/ComponentInitialize()
+	. = ..()
+	AddElement(/datum/element/flavor_text, _name = "Silicon Flavor Text", _save_key = "silicon_flavor_text")
+	AddElement(/datum/element/flavor_text, "", "Temporary Flavor Text", "This should be used only for things pertaining to the current round!")
+	AddElement(/datum/element/flavor_text, _name = "OOC Notes", _addendum = "Put information on ERP/vore/lewd-related preferences here. THIS SHOULD NOT CONTAIN REGULAR FLAVORTEXT!!", _always_show = TRUE, _save_key = "ooc_notes", _examine_no_preview = TRUE)
+
+
```
Comment (silicons) [inline code/modules/mob/living/silicon/silicon.dm]: no extra line here

File: modular_citadel/code/modules/client/preferences_savefile.dm [hunk 5]
```diff
@@ -35,4 +35,5 @@
 	WRITE_FILE(S["feature_xeno_head"], features["xenohead"])
 	//flavor text
 	WRITE_FILE(S["feature_flavor_text"], features["flavor_text"])
+	WRITE_FILE(S["silicon_feature_flavor_text"], features["silicon_flavor_text"])
```
Comment (silicons) [inline modular_citadel/code/modules/client/preferences_savefile.dm:38]: why is this here? put it where i put ooc nnotes sving/loading for characters
actually
@Ghommie can you take a look you touched savefiles last i'm a bit confused on what is going on regarding flavor_text vs feature_flavor_text

Comment (LetterN) [reply modular_citadel/code/modules/client/preferences_savefile.dm:38]: `feature_flavor_text` is the new var being used as a save key while `flavor_text` is the old one

File: code/modules/mob/living/silicon/silicon.dm (hunk 3 above)
Comment (Vizoplez) [reply code/modules/mob/living/silicon/silicon.dm]: fair

File: modular_citadel/code/modules/client/preferences_savefile.dm (hunk 5 above)
Comment (Vizoplez) [reply modular_citadel/code/modules/client/preferences_savefile.dm:38]: @kevinz000 honestly i had no other clue wherelse to put it, so i slapped it in the same place other flavortext was handled.

Comment (Poojawa) [reply modular_citadel/code/modules/client/preferences_savefile.dm:38]: old flavor text can probably be removed from the lists since I doubt anyone with characters from 2016 still exist

Also it was made a DNA feature as a lazy/easy fix for Changlings to not get metagame outed by having the wrong flavor text.
